### PR TITLE
Notification Entities + Swift Mk 1: Managed Object

### PR DESCRIPTION
**Pull request by jleandroperez**
Re-submitting PR #5831 (already validated + merged).

@kwonye mind taking a peek at this one?. 
Travis issues should be fixed. Long story short:

- I've managed to reproduce the failure locally
- It's not a bug introduced by the PR itself, but (somehow) it gets exposed!
- Tracked down to an unhandled NSException being thrown, while running the unit tests that don't even access Core Data
- Apparently related to the app being launched simultaneously while running (random) tests

I've added a try/catch block in the `TestContextHelper`, so that the exception gets logged, still, doesn't crash Travis.

cc @astralbodies  @sendhil

### CODE REVIEW

**Comment by kwonye** [comment]
Yea, this one is tricky because it _looks_ so unrelated, yet it so consistently passes/fails based on this PR.

@sendhil I'm not sure if cf3afa5 is deemed ok, do you mind giving your thoughts on this? It looks like it could just be masking the problem.

File: WordPress/WordPressTest/WPAvatarSourceTest.m [hunk 1]
```diff
@@ -48,7 +48,7 @@ - (void)testGravatarCaches
         XCTAssertNotNil(image, @"avatar should be downloaded");
         XCTAssertEqual(image.size.width, 48.f, @"avatar should be resized");
     }];
-    [self waitForExpectationsWithTimeout:5.0 handler:nil];
+    [self waitForExpectationsWithTimeout:30.0 handler:nil];
```
Comment (kwonye) [inline WordPress/WordPressTest/WPAvatarSourceTest.m]: 30 is a bit long. Can we go to 10 max?

Comment (jleandroperez) [inline WordPress/WordPressTest/WPAvatarSourceTest.m]: Actually, this was totally unrelated. Up in that class we're mocking the response, so... i'll revert this commit, thanks!

**Comment by jleandroperez** [comment]
cc @astralbodies

**Comment by jleandroperez** [comment]
@kwonye @astralbodies @sendhil Fixed, Travis is all green:

- Updated Unit Tests's **tearDown** method:
 - Added **mainContext.reset()** calls here and there
 - Explicitly removing the Core Data overrides
- Nuked Travis's Cache

Mind taking a peek?

Thanks gentlemen!!!

**Comment by sendhil** [comment]
I'm fine with the fix.

**Comment by jleandroperez** [comment]
Thanks Sendhil!

**Pull request merged**